pkg/vault/contracts/Router.sol

```diff
@@ -2,36 +2,28 @@
 
 pragma solidity ^0.8.24;
 
-import { IERC4626 } from "@openzeppelin/contracts/interfaces/IERC4626.sol";
-import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
 import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
-import { Address } from "@openzeppelin/contracts/utils/Address.sol";
 import { IPermit2 } from "permit2/src/interfaces/IPermit2.sol";
 
 import { IWETH } from "@balancer-labs/v3-interfaces/contracts/solidity-utils/misc/IWETH.sol";
 import { IRouter } from "@balancer-labs/v3-interfaces/contracts/vault/IRouter.sol";
 import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
 import "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
 
-import { RouterWethLib } from "./lib/RouterWethLib.sol";
-import { RouterCommon } from "./RouterCommon.sol";
+import { BaseRouter } from "./BaseRouter.sol";
 
 /**
- * @notice Entrypoint for swaps, liquidity operations, and corresponding queries.
- * @dev The external API functions unlock the Vault, which calls back into the corresponding hook functions.
- * These interact with the Vault, transfer tokens, settle accounting, and handle wrapping and unwrapping ETH.
+ * @notice Base contract for routing swaps and liquidity operations through custom logic.
+ * @dev Provides shared functionality for derived routers, such as managing approvals, performing token transfers,
+ * and interacting with external pools or Vaults. Designed to be extended with specific routing behavior.
  */
-contract Router is IRouter, RouterCommon {
-    using Address for address payable;
-    using RouterWethLib for IWETH;
-    using SafeCast for *;
-
+contract Router is IRouter, BaseRouter {
```

Comment:
This is fine, but I'm thinking it'd be great to preserve the `IRouter` interface as an all-in-one interface.

In other words, if I wanted to use the router for something I should be able to just import and interface and get the ball rolling. Same that happens with `IVault`, which is a wide net that catches everything.

Can we achieve the same here? Maybe if `IRouter` extends `IBaseRouter`?